main.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace csharp
+{
+    public static class StringHelper
+    {
+        public static string Reverse(this string input)
+        {
+            return new string(input.ToCharArray().Reverse().ToArray());
+        }
+    }
+
+    class Program
+    {
+        static int[] ScoreBetween(string x, string y)
+        {
+            int[] scores = new int[y.Length + 1];
+            int preScore = 0;
+            foreach(var xc in x)
+            {
+                preScore = 0;
+                for(int i = 0; i < y.Length; ++i)
+                {
+                    var yc = y[i];
+                    var tmp = scores[i + 1];
+                    scores[i + 1] = new int[] { xc == yc ? preScore + 1 : 0, scores[i], scores[i + 1] }.Max();
+                    preScore = tmp;
+                }
+            }
+            return scores;
+        }
+
+        static string GetLCS(string x, string y)
+        {
+            if (x.Length == 0 || y.Length == 0) return "";
+            if (x.Length == 1 && y.Contains(x)) return x;
+            if (y.Length == 1 && x.Contains(y)) return y;
```

Comment:
Please improve here like [this commit](https://github.com/c910335/LCS-Problem-in-Linear-Space/commit/793f1e550caead0cba0c76e80a860d2d0165bf2a).